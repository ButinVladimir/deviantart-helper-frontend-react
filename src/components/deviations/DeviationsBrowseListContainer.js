import { connect } from 'react-redux';
import DeviationsBrowseList from './DeviationsBrowseList';

/**
 * @description
 * Maps Redux state to header props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  deviations: state.deviations.deviationsBrowse,
});

export default connect(mapStateToProps)(DeviationsBrowseList);
