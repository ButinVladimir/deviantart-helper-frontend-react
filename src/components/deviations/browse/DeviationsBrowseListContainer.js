import { connect } from 'react-redux';
import DeviationsBrowseList from './DeviationsBrowseList';
import deviationsBrowseLoadPageActionCreator from '../../../redux/action-creators/deviations/browse/load-page';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  deviations: state.deviations.browse.deviations,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The dispatch.
 * @param {Object} ownProps - The own component props.
 * @returns {Object} Props.
 */
const mapDispatchToProps = (dispatch, ownProps) => ({
  preloadDeviationsHandler: () => dispatch(deviationsBrowseLoadPageActionCreator(ownProps.config)),
});

export default connect(mapStateToProps, mapDispatchToProps)(DeviationsBrowseList);
