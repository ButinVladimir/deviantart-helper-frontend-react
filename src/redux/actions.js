export const TOGGLE_MENU = 'TOGGLE_MENU';
export const MESSAGE_SHOW = 'MESSAGE_SHOW';
export const MESSAGE_HIDE = 'MESSAGE_HIDE';
export const ERROR_SHOW = 'ERROR_SHOW';
export const CLEAR_LOADED_DATA = 'CLEAR_LOADED_DATA';
export const LOCK_TOGGLE = 'LOCK_TOGGLE';

export const REVOKE = 'REVOKE';

export const USER_LOAD_INFO = 'USER_LOAD_INFO';

export const DEVIATIONS_BROWSE_CHANGE_FORM_FIELD_VALUES = 'DEVIATIONS_BROWSE_CHANGE_FORM_FIELD_VALUES';
export const DEVIATIONS_BROWSE_LOAD_PAGE = 'DEVIATIONS_BROWSE_LOAD_PAGE';

export const DEVIATIONS_DETAILS_SET_ID = 'DEVIATIONS_DETAILS_SET_ID';
export const DEVIATIONS_DETAILS_SET_DATA_LOCK_TOGGLE = 'DEVIATIONS_DETAILS_SET_DATA_LOCK_TOGGLE';
export const DEVIATIONS_DETAILS_SET_DATA = 'DEVIATIONS_DETAILS_SET_DATA';
export const DEVIATIONS_DETAILS_CHANGE_FORM_FIELD_VALUES = 'DEVIATIONS_DETAILS_CHANGE_FORM_FIELD_VALUES';
export const DEVIATIONS_DETAILS_CHANGE_TAB = 'DEVIATIONS_DETAILS_CHANGE_TAB';

export const DEVIATIONS_STATISTICS_CHANGE_TITLE = 'DEVIATIONS_STATISTICS_CHANGE_TITLE';
export const DEVIATIONS_STATISTICS_CHANGE_PUBLISHED_TIME_BEGIN = 'DEVIATIONS_STATISTICS_CHANGE_PUBLISHED_TIME_BEGIN';
export const DEVIATIONS_STATISTICS_CHANGE_PUBLISHED_TIME_END = 'DEVIATIONS_STATISTICS_CHANGE_PUBLISHED_TIME_END';
export const DEVIATIONS_STATISTICS_CHANGE_SORT_FIELD = 'DEVIATIONS_STATISTICS_CHANGE_SORT_FIELD';
export const DEVIATIONS_STATISTICS_CHANGE_SORT_ORDER = 'DEVIATIONS_STATISTICS_CHANGE_SORT_ORDER';
export const DEVIATIONS_STATISTICS_CHANGE_TIMESTAMP_BEGIN = 'DEVIATIONS_STATISTICS_CHANGE_TIMESTAMP_BEGIN';
export const DEVIATIONS_STATISTICS_CHANGE_TIMESTAMP_END = 'DEVIATIONS_STATISTICS_CHANGE_TIMESTAMP_END';
export const DEVIATIONS_STATISTICS_CHANGE_CHART_TYPE = 'DEVIATIONS_STATISTICS_CHANGE_CHART_TYPE';
export const DEVIATIONS_STATISTICS_LOAD_PAGE = 'DEVIATIONS_STATISTICS_LOAD_PAGE';

export const DEVIATIONS_CHART_CHANGE_FORM_FIELD_VALUES = 'DEVIATIONS_CHART_CHANGE_FORM_FIELD_VALUES';
