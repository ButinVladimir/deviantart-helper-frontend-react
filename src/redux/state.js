/**
 * @global
 * @description
 * State.
 *
 * @typedef {Object} State
 * @property {SharedState} shared - Shared state.
 * @property {UserState} user - User state.
 */

/**
 * @global
 * @description
 * State shared between pages.
 *
 * @typedef {Object} SharedState
 * @property {boolean} blocked - Is app blocked from user input.
 * @property {boolean} isLoggedIn - Is user logged in.
 */

/**
 * @global
 * @description
 * User state.
 *
 * @typedef {Object} UserState
 * @property {string} userName - User name.
 * @property {string} userId - User ID.
 * @property {string} userIcon - User icon.
 * @property {string} userType - User type.
 * @property {number} accessTokenExpires - Expiration date of access token.
 * @property {number} refreshTokenExpires - Expiration date of refresh token.
 */
