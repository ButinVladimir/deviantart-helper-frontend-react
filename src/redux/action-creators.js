import * as actions from './actions';

/**
 * @description
 * Creates action to load user info.
 *
 * @param {string} userId - The user ID.
 * @param {string} userName - The user name.
 * @param {string} userIcon - The user icon.
 * @param {string} userType - The user icon.
 * @param {number} accessTokenExpires - Expiration date of access token.
 * @param {number} refreshTokenExpires - Expiration date of refresh token.
 * @returns {LoadUserInfoAction} Action.
 */
export const createLoadUserInfoAction = (
  userId,
  userName,
  userIcon,
  userType,
  accessTokenExpires,
  refreshTokenExpires,
) => ({
  type: actions.LOAD_USER_INFO,
  userId,
  userName,
  userIcon,
  userType,
  accessTokenExpires,
  refreshTokenExpires,
});

/**
 * @description
 * Action to load user info.
 *
 * @typedef {Object} LoadUserInfoAction
 *
 * @param {string} userId - The user ID.
 * @param {string} userName - The user name.
 * @param {string} userIcon - The user icon.
 */

/**
 * @description
 * Creates action to browse deviations.
 *
 * @param {number} page - Page number.
 * @param {string} title - Title substring to find.
 * @param {number} publishedTimeStart - Deviation should be published after this time.
 * @param {number} publishedTimeEnd - Deviation should be published before this time.
 * @param {string} sort - Field to sort deviations by.
 * @param {number} order - Order to sort deviations by.
 * @returns {BrowseDeviationsAction} Action.
 */
export const createBrowseDevitaionsAction = (
  page,
  title,
  publishedTimeStart,
  publishedTimeEnd,
  sort,
  order,
) => ({
  type: actions.BROWSE_DEVIATIONS,
  page,
  title,
  publishedTimeStart,
  publishedTimeEnd,
  sort,
  order,
});

/**
 * @typedef {Object} BrowseDeviationsAction
 *
 * @property {number} page - Page number.
 * @property {string} title - Title substring to find.
 * @property {number} publishedTimeStart - Deviation should be published after this time.
 * @property {number} publishedTimeEnd - Deviation should be published before this time.
 * @property {string} sort - Field to sort deviations by.
 * @property {number} order - Order to sort deviations by.
 */
