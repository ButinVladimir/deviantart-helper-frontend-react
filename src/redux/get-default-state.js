import './state';

/**
 * @description
 * Creates default state.
 *
 * @returns {State} Default state.
 */
export default () => ({
  shared: {
    blocked: false,
    isLoggedIn: false,
  },
  user: {
    userName: '',
    userId: '',
    userIcon: '',
    userType: '',
    accessTokenExpires: 0,
    refreshTokenExpires: 0,
  },
});
