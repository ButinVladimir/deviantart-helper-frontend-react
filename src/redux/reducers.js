import getDefaultState from './get-default-state';
import * as actions from './actions';
import './state';

/**
 * @description
 * Shared state reducer.
 *
 * @param {SharedState} sharedState - Shared state.
 * @param {Object} action - The action.
 * @returns {SharedState} New shared state.
 */
const sharedReducer = (sharedState, action) => {
  const newState = Object.assign({}, sharedState);

  switch (action.type) {
    case actions.LOAD_USER_INFO:
      Object.assign(newState, {
        isLoggedIn: true,
      });
      break;

    default:
  }

  return newState;
};

/**
 * @description
 * User state reducer.
 *
 * @param {UserState} userState - User state.
 * @param {SharedState} sharedState - Shared state.
 * @param {Object} action - The action.
 * @returns {UserState} New user state.
 */
const userReducer = (userState, sharedState, action) => {
  const newState = Object.assign({}, userState);

  switch (action.type) {
    case actions.LOAD_USER_INFO:
      Object.assign(newState, {
        userName: action.userName,
        userId: action.userId,
        userIcon: action.userIcon,
        userType: action.userType,
        accessTokenExpires: action.accessTokenExpires,
        refreshTokenExpires: action.refreshTokenExpires,
      });
      break;

    default:
  }

  return newState;
};

/**
 * @description
 * Main reducer.
 *
 * @param {State} state - Current Redux state.
 * @param {Object} action - The action.
 * @returns {State} New Redux state.
 */
export default (state = getDefaultState(), action = {}) => ({
  shared: sharedReducer(state.shared, action),
  user: userReducer(state.user, state.shared, action),
});
