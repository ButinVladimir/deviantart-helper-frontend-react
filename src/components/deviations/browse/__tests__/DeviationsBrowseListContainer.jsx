import React from 'react';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import createDefaultStore from '../../../../redux/states/state';
import DeviationsBrowseListContainer from '../DeviationsBrowseListContainer';
import { DEVIATIONS_COMMON_TOGGLE_SELECTION } from '../../../../redux/actions';

describe('DeviationsBrowseListContainer', () => {
  const deviations = ['1', '2', '3', '4'].map(id => ({
    id,
    title: `Deviation ${id}`,
    url: `/path/to/deviation/${id}`,
    thumbnail: {
      height: 1,
      width: 2,
      src: `/path/to/thumbnail/${id}`,
    },
    publishedTime: 10,
    nsfw: false,
    views: 1,
    favourites: 2,
    comments: 3,
    downloads: 4,
  }));

  const state = {
    ...createDefaultStore(),
    deviations: {
      browse: {
        deviations,
      },
      common: {
        selectedIds: [],
      },
    },
  };

  it('can be rendered correctly', () => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(state);
    const wrapper = mount(
      <MemoryRouter>
        <Provider store={store}>
          <DeviationsBrowseListContainer />
        </Provider>
      </MemoryRouter>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can handle toggling selection', () => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(state);
    const wrapper = mount(
      <MemoryRouter>
        <Provider store={store}>
          <DeviationsBrowseListContainer />
        </Provider>
      </MemoryRouter>,
    );

    wrapper.find('input#selection-1').simulate('change');
    const actions = store.getActions();

    expect(actions).toEqual([
      { type: DEVIATIONS_COMMON_TOGGLE_SELECTION, id: '1' },
    ]);
  });
});
