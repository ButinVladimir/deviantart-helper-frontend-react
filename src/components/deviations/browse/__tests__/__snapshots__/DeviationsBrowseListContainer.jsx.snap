// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsBrowseListContainer can be rendered correctly 1`] = `
<section
  className="section"
  style={Object {}}
>
  <div
    className="container"
    style={Object {}}
  >
    <article
      className="media"
      style={Object {}}
    >
      <div
        className="image-container media-left"
        style={Object {}}
      >
        <div>
          <img
            alt="Deviation 1"
            src="/path/to/thumbnail/1"
          />
        </div>
        <div>
          <small
            className="has-text-grey"
          >
            1970-1-1 03:00:00
          </small>
        </div>
        <div>
          <a
            href="/path/to/deviation/1"
          >
            Visit DA page
          </a>
        </div>
      </div>
      <div
        className="media-content"
        style={Object {}}
      >
        <div
          className="content"
          style={Object {}}
        >
          <h1
            className="title is-3"
          >
            <a
              aria-current={null}
              href="/deviations/details/1"
              onClick={[Function]}
            >
              Deviation 1
            </a>
          </h1>
        </div>
        <div
          className="level is-tablet"
          style={Object {}}
        >
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  1
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Views
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  2
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Favourites
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  3
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Comments
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  4
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Downloads
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="media-right"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <div
            className="control"
            style={Object {}}
          >
            <label
              className="checkbox"
              disabled={false}
              style={Object {}}
            >
              <input
                checked={false}
                disabled={false}
                id="selection-1"
                name={null}
                onChange={[Function]}
                type="checkbox"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
    </article>
    <article
      className="media"
      style={Object {}}
    >
      <div
        className="image-container media-left"
        style={Object {}}
      >
        <div>
          <img
            alt="Deviation 2"
            src="/path/to/thumbnail/2"
          />
        </div>
        <div>
          <small
            className="has-text-grey"
          >
            1970-1-1 03:00:00
          </small>
        </div>
        <div>
          <a
            href="/path/to/deviation/2"
          >
            Visit DA page
          </a>
        </div>
      </div>
      <div
        className="media-content"
        style={Object {}}
      >
        <div
          className="content"
          style={Object {}}
        >
          <h1
            className="title is-3"
          >
            <a
              aria-current={null}
              href="/deviations/details/2"
              onClick={[Function]}
            >
              Deviation 2
            </a>
          </h1>
        </div>
        <div
          className="level is-tablet"
          style={Object {}}
        >
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  1
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Views
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  2
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Favourites
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  3
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Comments
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  4
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Downloads
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="media-right"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <div
            className="control"
            style={Object {}}
          >
            <label
              className="checkbox"
              disabled={false}
              style={Object {}}
            >
              <input
                checked={false}
                disabled={false}
                id="selection-2"
                name={null}
                onChange={[Function]}
                type="checkbox"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
    </article>
    <article
      className="media"
      style={Object {}}
    >
      <div
        className="image-container media-left"
        style={Object {}}
      >
        <div>
          <img
            alt="Deviation 3"
            src="/path/to/thumbnail/3"
          />
        </div>
        <div>
          <small
            className="has-text-grey"
          >
            1970-1-1 03:00:00
          </small>
        </div>
        <div>
          <a
            href="/path/to/deviation/3"
          >
            Visit DA page
          </a>
        </div>
      </div>
      <div
        className="media-content"
        style={Object {}}
      >
        <div
          className="content"
          style={Object {}}
        >
          <h1
            className="title is-3"
          >
            <a
              aria-current={null}
              href="/deviations/details/3"
              onClick={[Function]}
            >
              Deviation 3
            </a>
          </h1>
        </div>
        <div
          className="level is-tablet"
          style={Object {}}
        >
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  1
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Views
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  2
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Favourites
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  3
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Comments
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  4
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Downloads
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="media-right"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <div
            className="control"
            style={Object {}}
          >
            <label
              className="checkbox"
              disabled={false}
              style={Object {}}
            >
              <input
                checked={false}
                disabled={false}
                id="selection-3"
                name={null}
                onChange={[Function]}
                type="checkbox"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
    </article>
    <article
      className="media"
      style={Object {}}
    >
      <div
        className="image-container media-left"
        style={Object {}}
      >
        <div>
          <img
            alt="Deviation 4"
            src="/path/to/thumbnail/4"
          />
        </div>
        <div>
          <small
            className="has-text-grey"
          >
            1970-1-1 03:00:00
          </small>
        </div>
        <div>
          <a
            href="/path/to/deviation/4"
          >
            Visit DA page
          </a>
        </div>
      </div>
      <div
        className="media-content"
        style={Object {}}
      >
        <div
          className="content"
          style={Object {}}
        >
          <h1
            className="title is-3"
          >
            <a
              aria-current={null}
              href="/deviations/details/4"
              onClick={[Function]}
            >
              Deviation 4
            </a>
          </h1>
        </div>
        <div
          className="level is-tablet"
          style={Object {}}
        >
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  1
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Views
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  2
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Favourites
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  3
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Comments
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  4
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Downloads
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="media-right"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <div
            className="control"
            style={Object {}}
          >
            <label
              className="checkbox"
              disabled={false}
              style={Object {}}
            >
              <input
                checked={false}
                disabled={false}
                id="selection-4"
                name={null}
                onChange={[Function]}
                type="checkbox"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
    </article>
  </div>
</section>
`;
